WordPress/src/main/java/org/wordpress/android/ui/prefs/AppPrefs.java

```diff
@@ -777,7 +778,19 @@ public static void setImageOptimizeQuality(int quality) {
 
     public static int getImageOptimizeQuality() {
         int quality = getInt(DeletablePrefKey.IMAGE_OPTIMIZE_QUALITY, 0);
-        return quality > 1 ? quality : WPMediaUtils.OPTIMIZE_IMAGE_ENCODER_QUALITY;
+
+        // It's necessary to check that the quality int exists in the quality array in case of changes
+        // See #19644 for an example of when the array's values were changed
+        Context context = WordPress.getContext();
+        String[] validQualityValues = context.getResources().getStringArray(R.array.site_settings_image_quality_values);
+        boolean isQualityValid = Arrays.asList(validQualityValues).contains(String.valueOf(quality));
+
+        // If quality int does not exist in settings array, return the default quality value instead
+        if (!isQualityValid) {
+            setImageOptimizeQuality(WPMediaUtils.OPTIMIZE_IMAGE_ENCODER_QUALITY);
```

Comment:
In the case we previously persisted an invalid quality value (e.g. 85), since we are using the default value, I think we should explicitly return it. Otherwise, it will return the invalid `quality` value, which leads to the following state when tapping on _Image Quality_ option:

<img width="300" alt="Screenshot 2023-11-23 at 11 26 25" src="https://github.com/wordpress-mobile/WordPress-Android/assets/14905380/cab982b7-fac2-42e3-9655-8dd63ac11b80">

Reply:
```suggestion
            setImageOptimizeQuality(WPMediaUtils.OPTIMIZE_IMAGE_ENCODER_QUALITY);
            return WPMediaUtils.OPTIMIZE_IMAGE_ENCODER_QUALITY;
```

Reply:
Aha, good catch! 🤦‍♀️  

I've made sure we're returning the default value in https://github.com/wordpress-mobile/WordPress-Android/pull/19644/commits/acb9197254838ff873477d972cd5f1a214c0ef85.